### PR TITLE
Feature/lots-search

### LINKED ISSUE #278: JS error prevents top bar search engine working on some pages
Cames from [review on #273](https://github.com/eReuse/devicehub-teal/pull/273#pullrequestreview-979781523):

This JS error is raised on the following pages that prevents other JS based features to work (e.g. top bar search engine):
```js
main.js:221 Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')
    at main.js:221:50
    at main.js:353:3

// main.js:221
  /**
   * Avoid hide dropdown when user clicked inside
   */
  document.getElementById("dropDownLotsSelector").addEventListener("click", event => {
    event.stopPropagation();
  })
```
- [ ] /inventory/lot/add/
- [ ] /inventory/device/
- [ ] /inventory/upload-snapshot/
- [ ] /inventory/device/add/
- [ ] /labels/print
- [ ] /labels/

### PR DESCRIPTION
**Pull request by RubenPX**
This feature allow user learch lots when they will change lots in devices.

- [x] Implements [US 3334](https://tree.taiga.io/project/usody/us/3334)
- [x] Fixes #278

### CODE REVIEW

**Review by cayop — CHANGES_REQUESTED** [review]
All work fine.
But there are some urls than don't work. This urls is not a list of devices for than I'm not sure if we need implemented this feature for this urls. But this search is always visible.
I suggest that we either remove the search engine on the pages that don't work or make the search engine work on these urls.
This is the list of urls:
/inventory/lot/add/
/inventory/device/
/inventory/upload-snapshot/
/inventory/device/add/
/labels/print
/labels/

File: ereuse_devicehub/static/js/main_inventory.js [hunk 1]
```diff
@@ -317,6 +317,29 @@ function export_file(type_file) {
...
+                lot.parentElement.style.display = "";
+            } else {
+                lot.parentElement.style.display = "none";
+            }
+        }
+    }
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+    document.getElementById("lots-search").addEventListener("input", (e) => { lotsSearcher.doSearch(e.target.value) })
```
Comment (slamora) [inline ereuse_devicehub/static/js/main_inventory.js:341]: May this code fail if `document.getElementById("lots-search")` returns `null`?

See #278 for a related example.

File: ereuse_devicehub/templates/inventory/device_list.html [hunk 2]
```diff
@@ -445,5 +454,5 @@ <h5 class="card-title">Documents</h5>
     perPage: 20
   })
 </script>
-<script src="{{ url_for('static', filename='js/main_inventory.build.js') }}"></script>
+<script src="{{ url_for('static', filename='js/main_inventory.js') }}"></script>
```
Comment (slamora) [inline ereuse_devicehub/templates/inventory/device_list.html:457]: I guess this change is unintentional (revert it!)
```suggestion
<script src="{{ url_for('static', filename='js/main_inventory.build.js') }}"></script>
```

**Review by slamora — CHANGES_REQUESTED** [review]

File: ereuse_devicehub/static/js/main_inventory.js (hunk 1 above)
Comment (RubenPX) [reply ereuse_devicehub/static/js/main_inventory.js:341]: Fixed here -->191539608b2d7d560ddea64676b84e459af13800

**Pull request merged**